### ISSUE TITLE
Add option to overwrite/reupload existing photo

### ISSUE THREAD

**Issue by verymilan** [LycheeOrg/Lychee#1384]
I know that there already are plans for a versioning feature – yet my only dealbreaker for using Lychee in production seems to become the missing option to actually replace an image.

My usecase is to share albums with clients and friends for example after photographing an event. This means that i might share well over 100 photos and might revision a bigger number of them later – sometimes a quite short time later. Normally i would just drop them into my Nextcloud where they would be replaced. 
A workflow of deleting and reuploading a big batch of photos would not scale here. It would be amazing to have a third way for dealing with duplicates, other than actual duplicates or skipping the upload. :)

**Comment by ildyria** [comment]
**TL;DR: [...] replaced [...] would be [...] a third way for dealing with duplicates, other than actual duplicates or skipping the upload. :)**

**Comment by d7415** [comment]
> a third way for dealing with duplicates

I mentioned this in Gitter, but it's important to remember that to Lychee, they aren't duplicates if they've been edited.

- If you want to keep symlinks, you'd need a way to detect that one has changed, and which one(s).
- If not, it would be a way of matching "duplicates" based on filename instead of hash.

I think the former is more likely to be useful beyond this specific use case. I feel like we'd need to check the symlinks themselves for duplicate (or missing) targets to achieve this?

**Comment by verymilan** [comment]
I just want to mention that this feature request was not meant to be about symlinks... it might be related tho...

**Comment by d7415** [comment]
> I just want to mention that this feature request was not meant to be about symlinks... it might be related tho...

The only way I can think to achieve this otherwise would be on a per-photo basis, which I figured would be covered under "would not scale". Otherwise Lychee has no way of determining what should be replaced by what.

**Comment by ildyria** [comment]
> The only way I can think to achieve this otherwise would be on a per-photo basis, which I figured would be covered under "would not scale". Otherwise Lychee has no way of determining what should be replaced by what.

When uploading, check in current the pictures of the album if there is a name match. If so, overwrite would be the idea I think.

**Comment by d7415** [comment]
> When uploading, check in current the pictures of the album if there is a name match. If so, overwrite would be the idea I think.

Hmm.... Initially I dismissed that, but actually on an album level that sounds less bad.
